### PR TITLE
Post effect resize fixes

### PR DESCRIPTION
**Pull request by GSterbrant**
Resizing happens prior to rendering, only with respect to the input target, and resize is a private function, called internally by the effect.

### CODE REVIEW

**Comment by mvaligursky** [comment]
Perhaps also remove the resize getting called by the queue?

File: scripts/posteffects/posteffect-bloom.js [hunk 1]
```diff
@@ -179,17 +179,15 @@ function BloomEffect(graphicsDevice) {
...
 
 BloomEffect.prototype = Object.create(pc.PostEffect.prototype);
 BloomEffect.prototype.constructor = BloomEffect;
 
-BloomEffect.prototype.resize = function (target) {
+BloomEffect.prototype._resize = function (target) {
 
     var width, height;
-    if (target === null) {
+    if (target === undefined) {
```
Comment (mvaligursky) [inline scripts/posteffects/posteffect-bloom.js:190]: I'm not sure the target can be undefined here, we're inside rendering.

File: scripts/posteffects/posteffect-bloom.js [hunk 2]
```diff
@@ -179,23 +179,15 @@ function BloomEffect(graphicsDevice) {
...
 
-    var width, height;
-    if (target === null) {
-        width = this.device.width;
-        height = this.device.height;
-    } else {
-        width = target.colorBuffer.width;
-        height = target.colorBuffer.height;
-    }
+    const width = target.colorBuffer.width;
```
Comment (mvaligursky) [inline scripts/posteffects/posteffect-bloom.js:189]: no const .. just var .. this is ES5 still

**Review by mvaligursky — APPROVED** [review]

File: scripts/posteffects/posteffect-ssao.js [hunk 3]
```diff
@@ -377,23 +377,15 @@ function SSAOEffect(graphicsDevice, ssaoScript) {
...
 
-    var width, height;
-    if (target === null) {
-        width = Math.ceil(this.device.width / this.device.maxPixelRatio / this.downscale);
-        height = Math.ceil(this.device.height / this.device.maxPixelRatio / this.downscale);
-    } else {
-        width = Math.ceil(target.colorBuffer.width / this.device.maxPixelRatio / this.downscale);
-        height = Math.ceil(target.colorBuffer.height / this.device.maxPixelRatio / this.downscale);
-    }
+    var width = Math.ceil(target.colorBuffer.width / this.device.maxPixelRatio / this.downscale);
```
Comment (slimbuck) [inline scripts/posteffects/posteffect-ssao.js:387]: not sure maxPixelRatio should factor into this?

Comment (mvaligursky) [reply scripts/posteffects/posteffect-ssao.js:387]: Maks added this this in his PR and I think that's correct? If pixel ration says render to half the size screen res .. we should use half the size for internal buffers, unless `colorBuffer.width` already has this scale in?

Comment (mvaligursky) [reply scripts/posteffects/posteffect-ssao.js:387]: Looking at the code, it seems renderTargetScale is never modified nor publicly exposed / documented variable always holding 1. I suspect we should just remove it.

Comment (GSterbrant) [reply scripts/posteffects/posteffect-ssao.js:387]: I am guessing this is legacy code from trying to perform dynamic resolution downscaling? Just a speculation though.

Comment (slimbuck) [reply scripts/posteffects/posteffect-ssao.js:387]: Using maxPixelRatio here means the SSAO downscale buffer depends on the browser's CSS scaling rules, which I think is wrong.

SSAO downscaled buffer size should be a simple multiple of render target size.

**Pull request merged**